### ISSUE TITLE
General Support & QA Fixes

### ISSUE THREAD

**Issue by rsgonzal** [Iridescent-CM/technovation-app#3169]

**Comment by rsgonzal** [comment]
Idk if this error happens often, but just noting i've gotten it a couple times now but using if I run it again it passes.![Screen Shot 2021-12-01 at 8.33.13 PM.png](https://images.zenhubusercontent.com/60527169cde2772242d0de95/cebde0dd-7179-4cff-9525-7dc37a0c637b)

**Comment by shaunxp20** [comment]
> Idk if this error happens often, but just noting i've gotten it a couple times now but using if I run it again it passes.![Screen Shot 2021-12-01 at 8.33.13 PM.png](https://camo.githubusercontent.com/3e72fccc2e48e11ff988cabc1de7d87cf75c2854030de9e8b42fca78251e92e4/68747470733a2f2f696d616765732e7a656e68756275736572636f6e74656e742e636f6d2f3630353237313639636465323737323234326430646539352f63656264653064642d373137392d346366662d393532352d376463333761306336333762)

Yep, that's just a day in the life w/ our test suite - I have seen that one fail randomly too, I rerun it as well, and then it will pass. 🙄 🤯 🤕 🥺 Not sure it's worth investigating though b/c it's part of the old signup flow, so on the bright side, once we get to the cleanup phase, we won't need this test anymore! 😁 Things should get smoother/more stable as we go, I hope! 😅

**Comment by rsgonzal** [comment]
haha good to know! there was this one and another one that i was getting, but i'm sure you get that too 😅. sweeping it under the rug..

**Comment by dboyer** [comment]
On production, I registered, set my location, got parental consent, and then clicked from My Profile to the dashboard. The dashboard opened to /scores and showed this message which was confusing. Can it show build your team instead?

<img width="979" alt="Screen Shot 2021-12-02 at 4 41 06 PM" src="https://user-images.githubusercontent.com/516764/144507622-e1ec9826-e44d-4b08-b18e-0c57c52e94f9.png">

**Comment by rsgonzal** [comment]
> On production, I registered, set my location, got parental consent, and then clicked from My Profile to the dashboard. The dashboard opened to /scores and showed this message which was confusing. Can it show build your team instead?

This is because scores and certificates are open! And registration is technically closed. let me close scores and see what you get

**Comment by rsgonzal** [comment]
i converted roxana+mentormom@iridescentlearning.org from a student to a mentor and I can't login as them now 🤔

Not sure if it's a bigger issue, should probably be its own ticket!

**Comment by dboyer** [comment]
> > On production, I registered, set my location, got parental consent, and then clicked from My Profile to the dashboard. The dashboard opened to /scores and showed this message which was confusing. Can it show build your team instead?
> 
> This is because scores and certificates are open! And registration is technically closed. let me close scores and see what you get

Tried again this morning and all looks good. I was routed to "Build Your Team" on the dashboard and there was no survey option in the header.

**Comment by shaunxp20** [comment]
> * [ ]  i converted [roxana+mentormom@iridescentlearning.org](mailto:roxana+mentormom@iridescentlearning.org) from a student to a mentor and I can't login as them now 🤔

For this one, in the logs, I see: "Gender can't be blank"

Try setting that in the admin and see if that works. If that works, then I'm really puzzled as to how this worked before b/c I see the gender validation has been there since 2018. 🤨

**Comment by rsgonzal** [comment]
- [ ] having trouble logging in as a chapter ambassador too! The real problem was that ophelie was trying to decline the user as a chapter ambassador for this year and it didn't seem to be registering

**Comment by rsgonzal** [comment]
> For this one, in the logs, I see: "Gender can't be blank"
>Try setting that in the admin and see if that works. If that works, then I'm really puzzled as to how this worked before b/c I see the gender validation has been there since 2018. 🤨

This worked!! strange, probably the same should apply to summary & skills and interests 😅

**Comment by shaunxp20** [comment]
> ActionView::Template::Error: undefined method `signed?' for nil:NilClass

This one^ from an Airbrake alert should be fixed. Hopefully we won't run into it again. 🤞

**Comment by rsgonzal** [comment]
re: division filter
we're talking about here?
<img width="372" alt="Screen Shot 2021-12-03 at 1 22 26 PM" src="https://user-images.githubusercontent.com/22801026/144674674-4d1a3efb-67ca-4696-bc98-c32ef53c7c5c.png">

and when did we add age verification because i see 13 year olds 🤔
![screencapture-technovation-qa-herokuapp-admin-participants-2021-12-03-13_23_35](https://user-images.githubusercontent.com/22801026/144674734-b3d4a9a3-efe1-452f-a601-fc6c11985417.png)

**Comment by shaunxp20** [comment]
> re: division filter
> we're talking about here?

Yup!

> and when did we add age verification because i see 13 year olds, but actually should that matter 🤔

Looks like Nov 19, anything before that and you could have been any years of age. So yeah, QA data might be a little funky!

**Comment by shaunxp20** [comment]
> * [ ]  i converted [roxana+mentormom@iridescentlearning.org](mailto:roxana+mentormom@iridescentlearning.org) from a student to a mentor and I can't login as them now 🤔
> 
> 
> Not sure if it's a bigger issue, should probably be its own ticket!
> 
> edit: Based on the email though, i feel like i may have done some shady things with this account in the past. it's labeled as past student & returning mentor..
> 
> edit 2: just converted [roxana+s917@technovation.org](mailto:roxana+s917@technovation.org) and same thing!
> 
> edit 3: i'm in an endless loop of data terms agreement now, but only for this last account. i think this is when i didn't first sign in as the student and click the terms. as in i wasn't registered at all for the new season.
> 
> edit 4: third account i broke [roxana+666@technovation.org](mailto:roxana+666@technovation.org)

Not urgent, but what's the status of these and this issue about converting students to mentors? 🤔

**Comment by rsgonzal** [comment]
>Not urgent, but what's the status of these and this issue about converting students to mentors? 🤔

So it looks like:
- i convert a past student, who has not yet registered for the current season, to a mentor
- when i log in as that past student i am shown the data terms agreement
- i accept them and i get brought back to the data terms agreement and i am stuck in the loop

**Comment by shaunxp20** [comment]
> > Not urgent, but what's the status of these and this issue about converting students to mentors? 🤔
> 
> So it looks like:
> 
>     * i convert a past student, who has not yet registered for the current season, to a mentor
> 
>     * when i log in as that past student i am shown the data terms agreement
> 
>     * i accept them and i get brought back to the data terms agreement and i am stuck in the loop

I added a gender to roxana+s917@technovation.org and now I can login via impersonation. 😕 Have there been any successful student -> mentor conversions yet? 🤨

**Comment by shaunxp20** [comment]
Looking into this student -> mentor conversion issue some more... well, this is a little embarrassing... apparently this functionality changed earlier this year... by me! 😊 😅 I don't remember doing that, but I guess I did! 😀 It was changed in Feb this year, I wonder if there haven't been any conversions since then?! So we never ran into this issue until now?! 🤔 😅

**Comment by shaunxp20** [comment]
> I wonder if there haven't been any conversions since then?! So we never ran into this issue until now?! 🤔 😅

Or another possibility, we converted some, but they/we never logged in? 🤔 😖 

Nonetheless, good find! 🏆 🙌 I think we can provide a gender when a student is converted to a mentor, should we set this to the "Prefer not to say" option?

**Comment by rsgonzal** [comment]
> Nonetheless, good find! 🏆 🙌 I think we can provide a gender when a student is converted to a mentor, should we set this to the "Prefer not to say" option?

interesting it's still the gender! yeah i guess once you added that to (+s917), i can now login and do the data terms and move on. I've been logging in as them, not impersonating.

I think that's a good default! we can tell the program team to tell whoever they convert to complete their profiles.

I think maybe if we converted them in feb, it's possible they had already logged it at some point and done the data terms? so they weren't stuck in a loop? and then yeah maybe we never impersonated them. i feel like it's rare we do that

**Comment by shaunxp20** [comment]
> I think maybe if we converted them in feb, it's possible they had already logged it at some point and done the data terms? so they weren't stuck in a loop? 

Ah, I think I understand now! Trying save the data terms being agreed to - this is on the `account`, and the gender validation is on there as well, so it's trying to save this and the account isn't valid b/c of the missing gender. Just the perfect storm of having to agree to data terms! B/c if that's not there, then they can successfully log in, and they would run into this issue later when updating their account (and presumably) there would be a message telling them about needing to provide a gender.

Anywho, there's a fix for this, coming to QA soon! 😄

**Comment by shaunxp20** [comment]
While I was working on the SSO stuff, one thing led to another... and I realized our old registration flow has an integration to Mailgun to verify the email address before registering - where we send the email address (except for gmail ones) to Mailgun to see if it looks good (which is why we would get the "Sorry, the email address you typed appears to be invalid." messages when registering). Anywho, more just FYI, and maybe it's something we should incorporate or remove at some point.

**Comment by rsgonzal** [comment]
> While I was working on the SSO stuff, one thing led to another... and I realized our old registration flow has an integration to Mailgun to verify the email address before registering - where we send the email address (except for gmail ones) to Mailgun to see if it looks good (which is why we would get the "Sorry, the email address you typed appears to be invalid." messages when registering). Anywho, more just FYI, and maybe it's something we should incorporate or remove at some point.

so this isn't happening right now?

**Comment by shaunxp20** [comment]
> > While I was working on the SSO stuff, one thing led to another... and I realized our old registration flow has an integration to Mailgun to verify the email address before registering - where we send the email address (except for gmail ones) to Mailgun to see if it looks good (which is why we would get the "Sorry, the email address you typed appears to be invalid." messages when registering). Anywho, more just FYI, and maybe it's something we should incorporate or remove at some point.
> 
> so this isn't happening right now?

Correct, it's not happening now; we never incorporated this Mailgun service into the new registration flow.

**Comment by rsgonzal** [comment]
> edit 4: third account i broke roxana+666@technovation.org

this account was still broken (stuck in data terms loop) until i updated the mentor type in the admin

**Comment by shaunxp20** [comment]
> > edit 4: third account i broke [roxana+666@technovation.org](mailto:roxana+666@technovation.org)
> 
> this account was still broken (stuck in data terms loop) until i updated the mentor type in the admin

Ah, the fix was only for new student -> mentor conversions, but for any existing ones (w/o a gender) it would still be broken. If it's a high priority, we can create a script to update the existing ones.

**Comment by shaunxp20** [comment]
> > > edit 4: third account i broke [roxana+666@technovation.org](mailto:roxana+666@technovation.org)
> > 
> > 
> > this account was still broken (stuck in data terms loop) until i updated the mentor type in the admin
> 
> Ah, the fix was only for new student -> mentor conversions, but for any existing ones (w/o a gender) it would still be broken. If it's a high priority, we can create a script to update the existing ones.

Or maybe we can create a dataclip to see who's all affected. 🤔

**Comment by rsgonzal** [comment]
I don't think the above is a high priority. We will see if we get any errors and if we do, we have an admin solution.

**Comment by rsgonzal** [comment]
Here's something I found while testing. 

- as a beginner, tried signing up with an email but it was already taken
- went back a few screens because i had to edit the parent email
- went forward to the password screen and new email was there but i still had the errors
- but it was just the messages because i was still able to submit and get in
![Screen Shot 2021-12-07 at 9.47.34 PM.png](https://images.zenhubusercontent.com/60527169cde2772242d0de95/ed88e293-b59b-448f-90e1-f5d039674956)

**Comment by shaunxp20** [comment]
> Here's something I found while testing.
> 
>     * as a beginner, tried signing up with an email but it was already taken
> 
>     * went back a few screens because i had to edit the parent email
> 
>     * went forward to the password screen and new email was there but i still had the errors
> 
>     * but it was just the messages because i was still able to submit and get in
>       ![Screen Shot 2021-12-07 at 9.47.34 PM.png](https://camo.githubusercontent.com/676a34d7eca980470cf6421f5f80134d1765aafe0d87837721955e9627243fe7/68747470733a2f2f696d616765732e7a656e68756275736572636f6e74656e742e636f6d2f3630353237313639636465323737323234326430646539352f65643838653239332d623539622d343438662d393065312d663564303339363734393536)

Interesting. Good find! We had the thought to not even display this disabled field for parents, that would help w/ the confusion, and also this particular issue, I think. 🤔

**Comment by rsgonzal** [comment]
open #3183 for team photo and maybe this should go there, but posting here because of the error screen. I tried to change my profile pic (individual) and this might be because my photo is too large, but is this the error we've always shown?
![Screen Shot 2021-12-07 at 10.18.09 PM.png](https://images.zenhubusercontent.com/60527169cde2772242d0de95/0fda8e7e-71f9-4778-8535-eede92313734)

**Comment by dboyer** [comment]
On staging, I logged in as a judge with an account I created last month. It took me to a page where I had to agree to the data terms to proceed. After doing that, it redirected me to the mentor training where I had this pop-up box

<img width="767" alt="Screen Shot 2021-12-08 at 10 42 18 AM" src="https://user-images.githubusercontent.com/516764/145237838-ae814005-2878-45f1-8756-309942924e92.png">

This may be specific to staging, but I wanted to confirm the "this is the pop up" text wouldn't be on prod.

**Comment by dboyer** [comment]
@rsgonzal I registered on production, and there's a date error in the welcome email beginners receive when they register. It should list registration as opening in December 2021, not Dec 2020.

<img width="528" alt="Screen Shot 2021-12-08 at 10 45 55 AM" src="https://user-images.githubusercontent.com/516764/145238413-f12ecbb7-8dcf-4887-b330-489c9c0b82ec.png">

**Comment by rsgonzal** [comment]
> This may be specific to staging, but I wanted to confirm the "this is the pop up" text wouldn't be on prod.

this is my creativity! will definitely be edited on prod!

> @rsgonzal I registered on production, and there's a date error in the welcome email beginners receive when they register. It should list registration as opening in December 2021, not Dec 2020.

this is because the platform hasn't swiitched over! but you got this an hour ago? it should have switched by now..?

**Comment by shaunxp20** [comment]
> open #3183 for team photo and maybe this should go there, but posting here because of the error screen. I tried to change my profile pic (individual) and this might be because my photo is too large, but is this the error we've always shown? ![Screen Shot 2021-12-07 at 10.18.09 PM.png](https://camo.githubusercontent.com/b3d18657db7e1bf5ddb3e15bcb3169ddde25923e358cfefa03d8a1b7603b2f3a/68747470733a2f2f696d616765732e7a656e68756275736572636f6e74656e742e636f6d2f3630353237313639636465323737323234326430646539352f30666461386537652d373166392d343737382d383533352d656564653932333133373334)

Btw, the update on this one is that basically there is no update. 😞 Nothing confirmed - but I have a hunch this is related to the Rails upgrade. We're on a recent version of Rails, but our image upload libraries are using version that are from 2017. I tried updating these gems, but it's not simple b/c there are breaking changes, and we're using multiple libraries; and on top of this I don't know these libraries, or how it's all wired, or S3, or why we're getting this error... so, it's all adding up to being a very brain melting bug! 🤯 🤕 🤒 🥴

**Comment by rsgonzal** [comment]
> Btw, the update on this one is that basically there is no update. 😞 Nothing confirmed - but I have a hunch this is related to the Rails upgrade. We're on a recent version of Rails, but our image upload libraries are using version that are from 2017. I tried updating these gems, but it's not simple b/c there are breaking changes, and we're using multiple libraries; and on top of this I don't know these libraries, or how it's all wired, or S3, or why we're getting this error... so, it's all adding up to being a very brain melting bug! 🤯 🤕 🤒 🥴

hmm ok! this is affecting team photos and likely parts of submisisons as well. Reopening #3189

**Comment by shaunxp20** [comment]
Documenting some stuff for the image upload issue.

We were running into this weird utf8 issue, I found the fix (specifying "utf8" instead of "utf-8") here:
https://github.com/dwilkie/carrierwave_direct/issues/199

After that was addressed, we were running into some issues with our custom code regarding filename lengths, after I removed that, things seems to be working. Just FYI about this, if we run into any issues regarding file length.

Anyhow, it seems to be working now (it worked for me locally, and it's on it's way to QA now). And also, I wouldn't be surprised if we keep running into issues w/ uploads, we're running on ~5 year old libraries. 🥴
